src/job_cache/job_cache.cpp

```diff
@@ -196,6 +197,13 @@ wcl::optional<ConnectError> Cache::backoff_try_connect(int attempts) {
   wcl::xoshiro_256 rng(wcl::xoshiro_256::get_rng_seed());
   useconds_t backoff = 1000;
   for (int i = 0; i < attempts; i++) {
+    // We normally connect in about 3 tries on fresh connect so
+    // if we haven't connected at this point its a good spot to start
+    // start trying.
+    if (attempts > 3) {
```

Comment:
This should be `i` not `attempts`. Attempts never changes :)

also nit: maybe `(i + 1) % 3 == 0`?

Reply:
I didn't want to try just every 3rd time, as the time between invocations is going up exponentially, after a certain point we should just try again every-time. The 4th wait takes as long as all prior waits combined (in expectation).